spec/praxis-mapper/connection_factories/sequel_spec.rb

```diff
@@ -0,0 +1,56 @@
+require 'spec_helper'
+
+describe Praxis::Mapper::ConnectionFactories::Sequel do
+
+  let(:database) { Sequel.mock}
+  let(:thread) { Thread.current }
+  let(:connection_manager) { double('Praxis::Mapper::ConnectionManager', thread: thread) }
+
+  subject(:factory) { described_class.new(connection:database) }
+
+  context 'checkout' do
+    it 'returns the connection' do
+      factory.checkout(connection_manager).should be database
+    end
+
+    it 'allocates a connection to the thread of the manager' do
+      database.pool.allocated.should_not have_key(thread)
+      factory.checkout(connection_manager)
+      database.pool.allocated.should have_key(thread)
+    end
```

Comment:
we should add a case where it does NOT call the database.pool if the thread already owns the connection. Or perhaps that's in an "acquire" test?